src/Ledger/Deleg.lagda

```diff
@@ -98,10 +109,10 @@ requiredDeposit pp nothing = 0
 \begin{code}
 data _⊢_⇀⦇_,DELEG⦈_ : DelegEnv → DState → DCert → DState → Set where
   DELEG-delegate :
-    d ≡ requiredDeposit pp mc ⊔ requiredDeposit pp mc'
+    d ≡ requiredVDelegDeposit pp mv ⊔ requiredDeposit pp mc
```

Comment:
Since we need to switch the semantics of `nothing` to 'do nothing' for the delegations, we also need to change how the deposits work. Instead, we charge a deposit if `c` isn't already in `vDelegs` or `sDelegs`. You don't have to do this as part of this PR, I just wanted to point it out since your changes here will be removed then.